api/v1alpha1/opentelemetrycollector_types.go

```diff
@@ -51,6 +51,11 @@ type OpenTelemetryCollectorSpec struct {
 	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors=true
 	ServiceAccount string `json:"serviceAccount,omitempty"`
 
+	// SecurityContext will be set as the container security context
+	// +optional
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors=true
+	SecurityContext v1.SecurityContext `json:"securityContext,omitempty"`
```

Comment:
You have to use it here, don't you? 

https://github.com/open-telemetry/opentelemetry-operator/blob/aff74858930ceaddf8a972db455c716906f1bf39/pkg/collector/container.go#L71-L78

Reply:
Ah my bad. I thought it worked as expected in my tests but may be I didn't look hard enough. Let me spin up local openshfit and report back in 78 hours once it has started :)

Reply:
> in 78 hours once it has started

ROFLMAO

For the record, I use `crc` for local testing, and a cluster starts relatively quick (under 5 mins).

Reply:
Yeah, it does take roughly the same amount of time. I was just exaggerating. It's not that bad. It does shut down occasionally on it's own though and if i keep it running for longer than an hour, I start seeing a ton of i/o timeout network errors. Other than that it is pretty solid. I think I should be able to finish the test and update here in the next couple of hours or so if I don't get distracted. :)

Reply:
Fixed and verified on a KIND cluster locally. Thanks for catching this.